### DISCUSSION TITLE
Can I run all migrates once?

### DISCUSSION THREAD

**Discussion by arthur-fontaine** [prisma/prisma#10189] [Q&A] [answered]
Hello ! I'm using Prisma in a project, and due to an oversight on my part, the npm script was running the `prisma migrate` command every time, so all migrations were overwritten.
However, the `schema.prisma` file is included in my Git, so it is versioned. Can I get all its versions and run the `prisma migrate` command for each of them to get the final migration cleanly?

**Comment by TasinIshmam** [comment, marked as answer]
Hey @arthur-fontaine  :wave:  

I'm trying to understand your problem a bit more clearly. Basically, you had some valid migrations (let's say Migration A, B and C) but these were overwritten due to an oversight. 

> Can I get all its versions and run the prisma migrate command for each of them to get the final migration cleanly? 

You could just get the versioned `schema.prisma` files at your desired database state, and run `prisma migrate` for each of those states until all the changes/migrations have been applied as desired. Is there a specific challenge you're running into while doing this? 

If you could clarify these things I think I would understand a bit better  

**Q1**: Would you like to
    A) Reapply these migrations back again (So that your database now has three migrations applied, A, B and C). 
    OR 
    B) Apply one migration that contains the changes for all three (Apply a migration D with all the changes from  A,B,C). 

**Q2**: Additionally, is there any concern of data loss or would it be okay to overwrite the existing data in the database (I'm assuming it's a dev database). 

**Q3**: Were your desired migrations overwritten/deleted or did you accidentally apply new migrations on top?

**Reply by arthur-fontaine** [reply]
_A1_: I want to reapply these migrations back again.
_A2_: We can't overwrite the data. We deployed mistakenly in production...
_A3_: I applied new migrations on top.

However, I resolved my problem. I downloaded all the migrations and the database from my server, and recreated all migrations locally with all the versions of my `schema.prisma` file. My mistake (in the resolving of the problem) was that I thought that the migrations were overwritten if the commit message was the same.

Thanks for your answer!

**Reply by TasinIshmam** [reply]
Oh I see, happy to see it has been resolved :D 

I'll mark this as answered then.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
